smoke_tests/sns_create_topic_test.go

```diff
@@ -0,0 +1,256 @@
+package smoke_tests
+
+import (
+	"context"
+	"encoding/xml"
+	"net/http"
+	"testing"
+
+	"github.com/Admiral-Piett/goaws/app"
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/Admiral-Piett/goaws/app/utils"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/gavv/httpexpect/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_CreateTopicV1_json_success(t *testing.T) {
+	server := generateServer()
+	defer func() {
+		server.Close()
+		utils.ResetResources()
+	}()
+
+	// Target test
+	topicName := "new-topic-1"
+	sdkConfig, _ := config.LoadDefaultConfig(context.TODO())
+	sdkConfig.BaseEndpoint = aws.String(server.URL)
+	snsClient := sns.NewFromConfig(sdkConfig)
+	sdkResponse, err := snsClient.CreateTopic(context.TODO(), &sns.CreateTopicInput{
+		Name: &topicName,
+	})
+
+	// Should success
+	assert.Contains(t, *sdkResponse.TopicArn, topicName)
+	assert.Nil(t, err)
+
+	// Get created topic
+	listTopicsXML := struct {
+		Action  string `xml:"Action"`
+		Version string `xml:"Version"`
+	}{
+		Action:  "ListTopics",
+		Version: "2012-11-05",
+	}
+	e := httpexpect.Default(t, server.URL)
+	r := e.POST("/").
+		WithForm(listTopicsXML).
+		Expect().
+		Status(http.StatusOK).
+		Body().Raw()
+	r2 := app.ListTopicsResponse{}
+	xml.Unmarshal([]byte(r), &r2)
+	assert.Equal(t, 1, len(r2.Result.Topics.Member))
+	assert.Contains(t, r2.Result.Topics.Member[0].TopicArn, topicName)
+}
+
+func Test_CreateTopicV1_json_existant_topic(t *testing.T) {
+	server := generateServer()
+	defer func() {
+		server.Close()
+		utils.ResetResources()
+	}()
+
+	// Prepare existant topic
+	topicName := "new-topic-1"
+	sdkConfig, err := config.LoadDefaultConfig(context.TODO())
+	sdkConfig.BaseEndpoint = aws.String(server.URL)
+	snsClient := sns.NewFromConfig(sdkConfig)
+	sdkResponse, _ := snsClient.CreateTopic(context.TODO(), &sns.CreateTopicInput{
+		Name: &topicName,
+	})
+	assert.Contains(t, *sdkResponse.TopicArn, topicName)
+	assert.Nil(t, err)
+
+	// Target test: create topic with same name
+	sdkResponse, err = snsClient.CreateTopic(context.TODO(), &sns.CreateTopicInput{
+		Name: &topicName,
+	})
+
+	// Should success
+	assert.Contains(t, *sdkResponse.TopicArn, topicName)
+	assert.Nil(t, err)
+
+	// Topic should not be duplicated
+	listTopicsXML := struct {
+		Action  string `xml:"Action"`
+		Version string `xml:"Version"`
+	}{
+		Action:  "ListTopics",
+		Version: "2012-11-05",
+	}
+	e := httpexpect.Default(t, server.URL)
+	r := e.POST("/").
+		WithForm(listTopicsXML).
+		Expect().
+		Status(http.StatusOK).
+		Body().Raw()
+	r2 := app.ListTopicsResponse{}
+	xml.Unmarshal([]byte(r), &r2)
+	assert.Equal(t, 1, len(r2.Result.Topics.Member))
+	assert.Contains(t, r2.Result.Topics.Member[0].TopicArn, topicName)
+}
+
+func Test_CreateTopicV1_json_add_new_topic(t *testing.T) {
```

Comment:
What's the goal of this test?  To make sure we don't trip on the existing topics when we add more than one?  If so would `Test_CreateTopicV1_json_add_multiple_topics` be a more descriptive name?

Reply:
> To make sure we don't trip on the existing topics when we add more than one
Yes, exactly, that's the purpose. I will rename it!